docs/dev/dev_adr.md

```diff
@@ -108,4 +108,13 @@ There is a custom SoT Aggregation method, originally this pre-dated Nautobot Cor
 
 ## Git Actions
 
-The data source contract is used for reading from Git, but extended for pushing to Git. 
\ No newline at end of file
+The data source contract is used for reading from Git, but extended for pushing to Git.
+
+## Configuration Postprocessing
+
+Intended configuration generated by Golden Config Intended feature is designed for comparing to the "running" configuration (assuring compliance comparing to the backup configuration). Using the default Intended configuration for remediation/provisioning of a network device configuration is not always possible. For instance, no secrets should be rendered to the Intended configuration, as it is stored in Git/Database, or maybe some reordering of commands is required to create a valid configuration artifact.
+
+The PROCESSING feature, that is enabled as a Dynamic Application Feature, exposes a single device UI and API view to manipulate the Intended configuration available for Golden Config, with extra processing steps. There are some default functions (i.e. `render_secrets`), but the order can be changed or new methods can be inserted, via Nautobot configuration settings.
+
+!!! Note
```

Comment:
I was expecting to see here things like:

- there can be multiple functions it is processed by, and in order
- You mention single device at a time, but good to explain why, eg for fear of scaling 
- Only expose some of the jinja filters
- why the signature is the way it is

Reply:
resolved in f0a938b